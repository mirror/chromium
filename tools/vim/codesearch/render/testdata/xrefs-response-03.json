{
    "elapsed_ms": 33, 
    "xref_search_response": [
        {
            "status": 0, 
            "estimated_total_type_count": [
                {
                    "count": 6, 
                    "type": "OVERRIDDEN_BY", 
                    "type_id": 800
                }, 
                {
                    "count": 3, 
                    "type": "CALLED_AT", 
                    "type_id": 2300
                }, 
                {
                    "count": 1, 
                    "type": "HAS_DECLARATION", 
                    "type_id": 3300
                }
            ], 
            "grok_total_number_of_results": 10, 
            "grok_next_results_offset": 10, 
            "search_result": [
                {
                    "match": [
                        {
                            "line_number": 48, 
                            "type_id": "HAS_DECLARATION", 
                            "line_text": "  virtual int RestartWithAuth(const CompletionCallback& callback) = 0;", 
                            "grok_modifiers": {
                                "definition": false, 
                                "declaration": true, 
                                "autogenerated": false, 
                                "abstract": true, 
                                "global": false, 
                                "immutable": false, 
                                "private": false, 
                                "public": true, 
                                "terminal": false, 
                                "protected": false, 
                                "static": false, 
                                "virtual": true, 
                                "anonymous": false, 
                                "deprecated": false, 
                                "volatile": false, 
                                "constexpr_": false, 
                                "file_scoped": false
                            }, 
                            "signature": "cpp:net::class-ProxyClientSocket::RestartWithAuth(const base::Callback<void (int), base::internal::CopyMode::Copyable, base::internal::RepeatMode::Repeating> &)@chromium/../../net/http/proxy_client_socket.h|decl", 
                            "type": "HAS_DECLARATION"
                        }
                    ], 
                    "file": {
                        "name": "src/net/http/proxy_client_socket.h", 
                        "package_name": "chromium"
                    }
                }, 
                {
                    "match": [
                        {
                            "line_number": 66, 
                            "type_id": "OVERRIDDEN_BY", 
                            "line_text": "int HttpProxyClientSocket::RestartWithAuth(const CompletionCallback& callback) {", 
                            "grok_modifiers": {
                                "definition": true, 
                                "declaration": true, 
                                "autogenerated": false, 
                                "abstract": false, 
                                "global": false, 
                                "immutable": false, 
                                "private": false, 
                                "public": true, 
                                "terminal": false, 
                                "protected": false, 
                                "static": false, 
                                "virtual": true, 
                                "anonymous": false, 
                                "deprecated": false, 
                                "volatile": false, 
                                "constexpr_": false, 
                                "file_scoped": false
                            }, 
                            "signature": "cpp:net::class-HttpProxyClientSocket::RestartWithAuth(const base::Callback<void (int), base::internal::CopyMode::Copyable, base::internal::RepeatMode::Repeating> &)@chromium/../../net/http/http_proxy_client_socket.cc|def", 
                            "type": "OVERRIDDEN_BY"
                        }
                    ], 
                    "file": {
                        "name": "src/net/http/http_proxy_client_socket.cc", 
                        "package_name": "chromium"
                    }
                }, 
                {
                    "match": [
                        {
                            "line_number": 57, 
                            "type_id": "OVERRIDDEN_BY", 
                            "line_text": "  int RestartWithAuth(const CompletionCallback& callback) override;", 
                            "grok_modifiers": {
                                "definition": false, 
                                "declaration": true, 
                                "autogenerated": false, 
                                "abstract": false, 
                                "global": false, 
                                "immutable": false, 
                                "private": false, 
                                "public": true, 
                                "terminal": false, 
                                "protected": false, 
                                "static": false, 
                                "virtual": true, 
                                "anonymous": false, 
                                "deprecated": false, 
                                "volatile": false, 
                                "constexpr_": false, 
                                "file_scoped": false
                            }, 
                            "signature": "cpp:net::class-HttpProxyClientSocket::RestartWithAuth(const base::Callback<void (int), base::internal::CopyMode::Copyable, base::internal::RepeatMode::Repeating> &)@chromium/../../net/http/http_proxy_client_socket.h|decl", 
                            "type": "OVERRIDDEN_BY"
                        }
                    ], 
                    "file": {
                        "name": "src/net/http/http_proxy_client_socket.h", 
                        "package_name": "chromium"
                    }
                }, 
                {
                    "match": [
                        {
                            "line_number": 132, 
                            "type_id": "OVERRIDDEN_BY", 
                            "line_text": "int HttpProxyClientSocketWrapper::RestartWithAuth(", 
                            "grok_modifiers": {
                                "definition": true, 
                                "declaration": true, 
                                "autogenerated": false, 
                                "abstract": false, 
                                "global": false, 
                                "immutable": false, 
                                "private": false, 
                                "public": true, 
                                "terminal": false, 
                                "protected": false, 
                                "static": false, 
                                "virtual": true, 
                                "anonymous": false, 
                                "deprecated": false, 
                                "volatile": false, 
                                "constexpr_": false, 
                                "file_scoped": false
                            }, 
                            "signature": "cpp:net::class-HttpProxyClientSocketWrapper::RestartWithAuth(const base::Callback<void (int), base::internal::CopyMode::Copyable, base::internal::RepeatMode::Repeating> &)@chromium/../../net/http/http_proxy_client_socket_wrapper.cc|def", 
                            "type": "OVERRIDDEN_BY"
                        }, 
                        {
                            "line_number": 592, 
                            "type_id": "CALLED_AT", 
                            "line_text": "  return transport_socket_->RestartWithAuth(base::Bind(", 
                            "grok_modifiers": {}, 
                            "signature": "cpp:ticket_grok://chromium/../../net/http/http_proxy_client_socket_wrapper.cc:USAGE:20276(RestartWithAuth)-RANGE`usage", 
                            "type": "CALLED_AT"
                        }
                    ], 
                    "file": {
                        "name": "src/net/http/http_proxy_client_socket_wrapper.cc", 
                        "package_name": "chromium"
                    }
                }, 
                {
                    "match": [
                        {
                            "line_number": 86, 
                            "type_id": "OVERRIDDEN_BY", 
                            "line_text": "  int RestartWithAuth(const CompletionCallback& callback) override;", 
                            "grok_modifiers": {
                                "definition": false, 
                                "declaration": true, 
                                "autogenerated": false, 
                                "abstract": false, 
                                "global": false, 
                                "immutable": false, 
                                "private": false, 
                                "public": true, 
                                "terminal": false, 
                                "protected": false, 
                                "static": false, 
                                "virtual": true, 
                                "anonymous": false, 
                                "deprecated": false, 
                                "volatile": false, 
                                "constexpr_": false, 
                                "file_scoped": false
                            }, 
                            "signature": "cpp:net::class-HttpProxyClientSocketWrapper::RestartWithAuth(const base::Callback<void (int), base::internal::CopyMode::Copyable, base::internal::RepeatMode::Repeating> &)@chromium/../../net/http/http_proxy_client_socket_wrapper.h|decl", 
                            "type": "OVERRIDDEN_BY"
                        }
                    ], 
                    "file": {
                        "name": "src/net/http/http_proxy_client_socket_wrapper.h", 
                        "package_name": "chromium"
                    }
                }, 
                {
                    "match": [
                        {
                            "line_number": 81, 
                            "type_id": "OVERRIDDEN_BY", 
                            "line_text": "int SpdyProxyClientSocket::RestartWithAuth(const CompletionCallback& callback) {", 
                            "grok_modifiers": {
                                "definition": true, 
                                "declaration": true, 
                                "autogenerated": false, 
                                "abstract": false, 
                                "global": false, 
                                "immutable": false, 
                                "private": false, 
                                "public": true, 
                                "terminal": false, 
                                "protected": false, 
                                "static": false, 
                                "virtual": true, 
                                "anonymous": false, 
                                "deprecated": false, 
                                "volatile": false, 
                                "constexpr_": false, 
                                "file_scoped": false
                            }, 
                            "signature": "cpp:net::class-SpdyProxyClientSocket::RestartWithAuth(const base::Callback<void (int), base::internal::CopyMode::Copyable, base::internal::RepeatMode::Repeating> &)@chromium/../../net/spdy/spdy_proxy_client_socket.cc|def", 
                            "type": "OVERRIDDEN_BY"
                        }
                    ], 
                    "file": {
                        "name": "src/net/spdy/spdy_proxy_client_socket.cc", 
                        "package_name": "chromium"
                    }
                }, 
                {
                    "match": [
                        {
                            "line_number": 62, 
                            "type_id": "OVERRIDDEN_BY", 
                            "line_text": "  int RestartWithAuth(const CompletionCallback& callback) override;", 
                            "grok_modifiers": {
                                "definition": false, 
                                "declaration": true, 
                                "autogenerated": false, 
                                "abstract": false, 
                                "global": false, 
                                "immutable": false, 
                                "private": false, 
                                "public": true, 
                                "terminal": false, 
                                "protected": false, 
                                "static": false, 
                                "virtual": true, 
                                "anonymous": false, 
                                "deprecated": false, 
                                "volatile": false, 
                                "constexpr_": false, 
                                "file_scoped": false
                            }, 
                            "signature": "cpp:net::class-SpdyProxyClientSocket::RestartWithAuth(const base::Callback<void (int), base::internal::CopyMode::Copyable, base::internal::RepeatMode::Repeating> &)@chromium/../../net/spdy/spdy_proxy_client_socket.h|decl", 
                            "type": "OVERRIDDEN_BY"
                        }
                    ], 
                    "file": {
                        "name": "src/net/spdy/spdy_proxy_client_socket.h", 
                        "package_name": "chromium"
                    }
                }, 
                {
                    "match": [
                        {
                            "line_number": 288, 
                            "type_id": "CALLED_AT", 
                            "line_text": "        return proxy_socket->RestartWithAuth(connect_callback_);", 
                            "grok_modifiers": {}, 
                            "signature": "cpp:ticket_grok://chromium/../../jingle/glue/proxy_resolving_client_socket.cc:USAGE:11568(RestartWithAuth)-RANGE`usage", 
                            "type": "CALLED_AT"
                        }
                    ], 
                    "file": {
                        "name": "src/jingle/glue/proxy_resolving_client_socket.cc", 
                        "package_name": "chromium"
                    }
                }, 
                {
                    "match": [
                        {
                            "line_number": 1295, 
                            "type_id": "CALLED_AT", 
                            "line_text": "  return proxy_socket->RestartWithAuth(io_callback_);", 
                            "grok_modifiers": {}, 
                            "signature": "cpp:ticket_grok://chromium/../../net/http/http_stream_factory_impl_job.cc:USAGE:47447(RestartWithAuth)-RANGE`usage", 
                            "type": "CALLED_AT"
                        }
                    ], 
                    "file": {
                        "name": "src/net/http/http_stream_factory_impl_job.cc", 
                        "package_name": "chromium"
                    }
                }
            ]
        }
    ]
}
