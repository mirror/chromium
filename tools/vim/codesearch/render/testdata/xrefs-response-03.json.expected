This file contains the rendered output per line and its associated metadata.

Lines that begin with a line number contains (after the '|') the contents that
will be inserted into the vim buffer at that line. The immediately following
line contains an object representing associated metadata.
-------------------------------------------------------------------------------

001|^Cat{Declaration:
   |{}
002|}Cat_  ^F{src/net/http/proxy_client_socket.h}F_
   |{'j': ('src/net/http/proxy_client_socket.h', 0, 7)}
003|    48:  ^>{  virtual int RestartWithAuth(const CompletionCallback& callback) = 0;}>_
   |{'j': ('src/net/http/proxy_client_socket.h', 47, 9)}
004|
   |{}
005|^Cat{Calls:
   |{}
006|}Cat_  ^F{src/jingle/glue/proxy_resolving_client_socket.cc}F_
   |{'j': ('src/jingle/glue/proxy_resolving_client_socket.cc', 0, 7)}
007|    288:  ^>{        return proxy_socket->RestartWithAuth(connect_callback_);}>_
   |{'j': ('src/jingle/glue/proxy_resolving_client_socket.cc', 287, 10)}
008|
   |{}
009|  ^F{src/net/http/http_proxy_client_socket_wrapper.cc}F_
   |{'j': ('src/net/http/http_proxy_client_socket_wrapper.cc', 0, 2)}
010|    592:  ^>{  return transport_socket_->RestartWithAuth(base::Bind(}>_
   |{'j': ('src/net/http/http_proxy_client_socket_wrapper.cc', 591, 10)}
011|
   |{}
012|  ^F{src/net/http/http_stream_factory_impl_job.cc}F_
   |{'j': ('src/net/http/http_stream_factory_impl_job.cc', 0, 2)}
013|    1295:  ^>{  return proxy_socket->RestartWithAuth(io_callback_);}>_
   |{'j': ('src/net/http/http_stream_factory_impl_job.cc', 1294, 11)}
014|
   |{}
015|^Cat{Overridden by:
   |{}
016|}Cat_  ^F{src/net/http/http_proxy_client_socket.cc}F_
   |{'j': ('src/net/http/http_proxy_client_socket.cc', 0, 7)}
017|    66:  ^>{int HttpProxyClientSocket::RestartWithAuth(const CompletionCallback& callback) {}>_
   |{'j': ('src/net/http/http_proxy_client_socket.cc', 65, 9)}
018|
   |{}
019|  ^F{src/net/http/http_proxy_client_socket.h}F_
   |{'j': ('src/net/http/http_proxy_client_socket.h', 0, 2)}
020|    57:  ^>{  int RestartWithAuth(const CompletionCallback& callback) override;}>_
   |{'j': ('src/net/http/http_proxy_client_socket.h', 56, 9)}
021|
   |{}
022|  ^F{src/net/http/http_proxy_client_socket_wrapper.cc}F_
   |{'j': ('src/net/http/http_proxy_client_socket_wrapper.cc', 0, 2)}
023|    132:  ^>{int HttpProxyClientSocketWrapper::RestartWithAuth(}>_
   |{'j': ('src/net/http/http_proxy_client_socket_wrapper.cc', 131, 10)}
024|
   |{}
025|  ^F{src/net/http/http_proxy_client_socket_wrapper.h}F_
   |{'j': ('src/net/http/http_proxy_client_socket_wrapper.h', 0, 2)}
026|    86:  ^>{  int RestartWithAuth(const CompletionCallback& callback) override;}>_
   |{'j': ('src/net/http/http_proxy_client_socket_wrapper.h', 85, 9)}
027|
   |{}
028|  ^F{src/net/spdy/spdy_proxy_client_socket.cc}F_
   |{'j': ('src/net/spdy/spdy_proxy_client_socket.cc', 0, 2)}
029|    81:  ^>{int SpdyProxyClientSocket::RestartWithAuth(const CompletionCallback& callback) {}>_
   |{'j': ('src/net/spdy/spdy_proxy_client_socket.cc', 80, 9)}
030|
   |{}
031|  ^F{src/net/spdy/spdy_proxy_client_socket.h}F_
   |{'j': ('src/net/spdy/spdy_proxy_client_socket.h', 0, 2)}
032|    62:  ^>{  int RestartWithAuth(const CompletionCallback& callback) override;}>_
   |{'j': ('src/net/spdy/spdy_proxy_client_socket.h', 61, 9)}
033|
   |{}
034|
   |{}
